secrets/errors.go

```diff
@@ -9,6 +9,9 @@ import (
 // encoding in the secrets.json file.
 var ErrInvalidEncoding = errors.New("secrets: invalid encoding, expected identity, base64 or empty")
 
+// ErrEmptySecretPath is returned when the key for a secret is empty.
+var ErrEmptySecretPath = errors.New("secrets: secret path cannot be empty")
```

Comment:
Maybe this should be "key" rather than "path"?  Path sounds like the path to the secrets file IMO.

Reply:
Agreed, as I was re-reading it I was thinking of the file path.

Reply:
we do call them "secret path" in some context. our terminology is confusing 😞

Reply:
I did rename the parameter of those methods too.